File: Lab6/nQueen.py
def queens_placement(n):
    board = [[0] * n for _ in range(n)]
    place_queens(0, board, n)

def place_queens(row, board, n):
    if row == n:
        for line in board:
            print(" ".join(map(str, line)))  # Fixed missing string before .join()
        print()
        return

    for col in range(n):
        if safe_check(board, row, col, n):
            board[row][col] = 1
            place_queens(row + 1, board, n)
            board[row][col] = 0  # Backtrack

def safe_check(board, x, y, n):
    for i in range(x):  # Check column
        if board[i][y] == 1:
            return False

    i, j = x - 1, y - 1  # Check upper left diagonal
    while i >= 0 and j >= 0:
        if board[i][j] == 1:
            return False
        i -= 1
        j -= 1

    i, j = x - 1, y + 1  # Check upper right diagonal
    while i >= 0 and j < n:
        if board[i][j] == 1:
            return False
        i -= 1
        j += 1
    
    return True

if __name__ == "__main__":
    n = int(input("Enter the number of queens: "))  # Fixed syntax
    queens_placement(n)

File: Lab6/nqueen.js
function queensPlacement(n) {
    let board = Array.from({ length: n }, () => Array(n).fill(0));
    placeQueens(0, board, n);
}

function placeQueens(row, board, n) {
    if (row === n) {
        board.forEach(line => console.log(line.join(" ")));
        console.log();
        return;
    }

    for (let col = 0; col < n; col++) {
        if (safeCheck(board, row, col, n)) {
            board[row][col] = 1;
            placeQueens(row + 1, board, n);
            board[row][col] = 0; // Backtrack
        }
    }
}

function safeCheck(board, x, y, n) {
    for (let i = 0; i < x; i++) {
        if (board[i][y] === 1) return false;
    }

    for (let i = x - 1, j = y - 1; i >= 0 && j >= 0; i--, j--) {
        if (board[i][j] === 1) return false;
    }

    for (let i = x - 1, j = y + 1; i >= 0 && j < n; i--, j++) {
        if (board[i][j] === 1) return false;
    }

    return true;
}

const n = parseInt(prompt("Enter the number of queens: "), 10);
queensPlacement(n);
